api/src/test/java/au/org/aodn/nrmn/restapi/service/SpreadSheetServiceIT.java

```diff
@@ -230,4 +230,18 @@ void duplicateRowsShouldBeRemoved() throws Exception {
         assertEquals(20, validRows.size());
     }
 
+    /** This test is for:
+     * When adding new jobs, only rows (with "total=0" and duplicate more than 3 times) at the end of the sheet should be
+     * removed. Rows (with "total=0" and duplicate more than 3 times) are not at the end of the sheet should be kept.
+     * @throws Exception
+     */
+    @Test
+    void removeBottomDuplicateRowsOnly() throws Exception {
+        FileSystemResource file = new FileSystemResource("src/test/resources/sheets/removeDuplicateBottomRows.xlsx");
+        MockMultipartFile mockFile = new MockMultipartFile("sheets/removeDuplicateBottomRows.xlsx", file.getInputStream());
```

Comment:
The removeDuplicateBottomRows contains 495 lines, the last 7 line is total=0 and repeat more then 3 times,  the first two line are header, so 495 - 7 - 2 = 486 so why 487 is expected result?

Reply:
to record after discussion: if the last row is valid (species is "SURVEY NOT DONE", "DEBRIS - ZERO", or "NO SPECIES FOUND"), it will keep this row. Need to add a new test for it (valid, keep it. Otherwise remove it).